File: 3.4/python/3.4.py
class hanoi_class():
    count = 0
    def move(self, x, n, z):
        print "Move %d from %c to %c" % (n, x, z)
 
    def hanoi(self, n, x, y, z):
        if n==1:
            self.move(x, 1, z)
        else:
            self.hanoi(n-1, x, z, y)
            self.move(x, n, z)
            self.hanoi(n-1, y, x, z)
 
if __name__ == "__main__":
    s = hanoi_class()
    s.hanoi(4, 'a', 'b', 'c');

File: chap3/3.4/cpp/3.4.cpp
#include <iostream>
using namespace std;

void move(char x, int n, char z) // move n from x to z
{
        static int count = 0;
        cout << ++count << ": move " << n << " from " << x << " to " << z << endl;
}

void hanoi(int n, char x, char y, char z)
{
        if (n == 1) 
        {
                move(x, 1, z);
        }
        else 
        {
                hanoi(n-1, x, z, y);
                move(x, n, z);
                hanoi(n-1, y, x, z);
        }
}

int main()
{
        hanoi(4, 'a', 'b', 'c');
}

